# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.71; am-1.16; lt-2.4.6)
libtoolize -c -f; aclocal; autoconf; autoheader; automake -a -c
./configure --disable-static --enable-shared --prefix=

# build
make

# install
install -d $DESTDIR/bin
install src/extract_a52 $DESTDIR/bin
install src/.libs/a52dec $DESTDIR/bin
install -d $DESTDIR/include/a52dec
install -m 644 include/a52.h $DESTDIR/include/a52dec
install -m 644 include/mm_accel.h $DESTDIR/include/a52dec
install -d $DESTDIR/lib
install liba52/.libs/liba52.so.0.* $DESTDIR/lib/liba52.so.0
ln -sf liba52.so.0 $DESTDIR/lib/liba52.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 liba52/liba52.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/a52dec-build.pc << EOF
Name: a52dec
Version: 0.8.0
Description: ATSC A/52 (AC-3) decoder library
Requires: $REQUIRES

devel=\\
/include/a52dec/a52.h \\
/include/a52dec/mm_accel.h \\
/lib/liba52.so \\
/lib/pkgconfig/liba52.pc

exec=\\
/bin/a52dec \\
/bin/extract_a52 \\
/lib/liba52.so.0
EOF

exit
--- a52dec-0.8.0.orig/acinclude.m4
+++ a52dec-0.8.0/acinclude.m4
@@ -6,7 +6,7 @@
     [AC_MSG_CHECKING([for restrict])
     ac_cv_c_restrict=no
     for ac_kw in restrict __restrict__ __restrict; do
-	AC_TRY_COMPILE([],[char * $ac_kw p;],[ac_cv_c_restrict=$ac_kw; break])
+	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[char * $ac_kw p;])],[ac_cv_c_restrict=$ac_kw; break])
     done
     AC_MSG_RESULT([$ac_cv_c_restrict])
     case $ac_cv_c_restrict in
@@ -30,7 +30,7 @@
 }
 EOF
 	if AC_TRY_COMMAND([${CC-cc} $CFLAGS -nostdlib -nostartfiles
-            -o conftest conftest.c -lgcc >&AC_FD_CC]); then
+            -o conftest conftest.c -lgcc >&AS_MESSAGE_LOG_FD]); then
 	    ac_cv_builtin_expect=yes
 	else
 	    ac_cv_builtin_expect=no
@@ -50,11 +50,11 @@
 	AC_MSG_CHECKING([for always_inline])
 	SAVE_CFLAGS="$CFLAGS"
 	CFLAGS="$CFLAGS -Wall -Werror"
-	AC_TRY_COMPILE([],
+	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],
 	    [__attribute__ ((__always_inline__)) void f (void);
 	    #ifdef __cplusplus
 	    42 = 42;	// obviously illegal - we want c++ to fail here
-	    #endif],
+	    #endif])],
 	    [ac_cv_always_inline=yes],[ac_cv_always_inline=no])
 	CFLAGS="$SAVE_CFLAGS"
 	AC_MSG_RESULT([$ac_cv_always_inline])
@@ -72,8 +72,8 @@
 	[ac_cv_c_attribute_aligned],
 	[ac_cv_c_attribute_aligned=0
 	for ac_cv_c_attr_align_try in 2 4 8 16 32 64; do
-	    AC_TRY_COMPILE([],
-		[static char c __attribute__ ((aligned($ac_cv_c_attr_align_try))) = 0; return c;],
+	    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],
+		[static char c __attribute__ ((aligned($ac_cv_c_attr_align_try))) = 0; return c;])],
 		[ac_cv_c_attribute_aligned=$ac_cv_c_attr_align_try])
 	done])
     if test x"$ac_cv_c_attribute_aligned" != x"0"; then
@@ -88,7 +88,7 @@
     [AC_MSG_CHECKING([if $CC supports $1 flags])
     SAVE_CFLAGS="$CFLAGS"
     CFLAGS="$1"
-    AC_TRY_COMPILE([],[],[ac_cv_try_cflags_ok=yes],[ac_cv_try_cflags_ok=no])
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[])],[ac_cv_try_cflags_ok=yes],[ac_cv_try_cflags_ok=no])
     CFLAGS="$SAVE_CFLAGS"
     AC_MSG_RESULT([$ac_cv_try_cflags_ok])
     if test x"$ac_cv_try_cflags_ok" = x"yes"; then
@@ -172,7 +172,7 @@
 old_CXXFLAGS="$CXXFLAGS" ; CXXFLAGS=""
 old_CPPFLAGS="$CPPFLAGS" ; CPPFLAGS=""
 old_CFLAGS="$CFLAGS"     ; CFLAGS=""
-AC_TRY_COMPILE([#include <stdint.h>],[int_least32_t v = 0;],
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([#include <stdint.h>],[int_least32_t v = 0;])],
 [ac_cv_stdint_result="(assuming C99 compatible system)"
  ac_cv_header_stdint_t="stdint.h"; ],
 [ac_cv_header_stdint_t=""])
--- a52dec-0.8.0.orig/configure.ac
+++ a52dec-0.8.0/configure.ac
@@ -3,6 +3,7 @@
 AC_INIT([a52dec],[0.8.0])
 AC_CONFIG_SRCDIR([src/a52dec.c])
 AC_CONFIG_AUX_DIR(autotools)
+AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_FILES([Makefile autotools/Makefile include/Makefile test/Makefile
     doc/Makefile src/Makefile liba52/Makefile libao/Makefile vc++/Makefile
     liba52/liba52.pc])
@@ -86,15 +87,13 @@
 
 dnl Checks for libtool - this must be done after we set cflags
 AC_DISABLE_SHARED
-AC_LIBTOOL_WIN32_DLL
-AC_PROG_LIBTOOL
+LT_INIT
 
 dnl Checks for libraries.
 
 dnl Checks for header files.
 AM_CPPFLAGS='-I$(top_srcdir)/include -I$(top_builddir)/include'
 AC_SUBST([AM_CPPFLAGS])
-AC_HEADER_TIME
 AC_CHECK_TYPES([struct timeval], [], [],
     [#include <sys/time.h>
     #include <time.h>])
@@ -107,7 +106,6 @@
 AC_C_BUILTIN_EXPECT
 AC_C_BIGENDIAN
 AC_TYPE_SIZE_T
-AC_TYPE_SIGNAL
 AC_SYS_LARGEFILE
 
 dnl Checks for library functions.
--- a52dec-0.8.0.orig/src/a52dec.c
+++ a52dec-0.8.0/src/a52dec.c
@@ -60,11 +60,10 @@
 
 static void print_fps (int final);
 
-static RETSIGTYPE signal_handler (int sig)
+static void signal_handler (int sig)
 {
     sigint = 1;
     signal (sig, SIG_DFL);
-    return (RETSIGTYPE)0;
 }
 
 static void print_fps (int final)
